spec/lib/sponsor_users_spec.rb

```diff
@@ -0,0 +1,183 @@
+    let(:sponsees) { ['adrian@example.com'] }
+
+    it 'Calls user_model#generate with the sponsees email' do
+      expect(user_model).to have_received(:generate) \
+        .with(contact: 'adrian@example.com', sponsor: 'chris@gov.uk')
+    end
+
+    it 'Sends an email to the sponsee_address with the login details' do
+      expect(a_signup_email_request(email: 'adrian@example.com')).to have_been_made.once
+    end
+
+    it 'Sends a single user confirmation email to the sponsor' do
+      body = {
+        email_address: 'chris@gov.uk',
+        template_id: '30ab6bc5-20bf-45af-b78d-34cacc0027cd',
+        personalisation: {
+          contact: sponsees.first
+        }
+      }
+      expect(a_request(:post, notify_email_url).with(notify_request(body))).to have_been_made.once
+    end
+  end
+
+  context 'Sponsoring a single phone number' do
+    let(:sponsor) { 'Craig <craig@gov.uk>' }
+    let(:sponsees) { ['+447700900003'] }
+
+    it 'Calls user_model#generate with the sponsees phone number' do
+      expect(user_model).to have_received(:generate) \
+        .with(contact: '+447700900003', sponsor: 'craig@gov.uk')
+    end
+
+    it 'Sends an sms to the sponsee_address confirming the signup' do
+      expect(a_signup_sms_request(phone_number: '+447700900003')).to have_been_made.once
+    end
+  end
+
+  context 'Sponsoring the same phone number twice' do
+    let(:sponsor) { 'Craig <craig@gov.uk>' }
+    let(:sponsees) { ['+447700900003', '+447700900003'] }
+
+    it 'Calls user_model#generate once' do
+      expect(user_model).to have_received(:generate) \
+        .with(contact: '+447700900003', sponsor: 'craig@gov.uk').once
+    end
+  end
+
+  context 'Sponsoring an email address and a phone number' do
+    let(:sponsor) { 'Chloe <chloe@gov.uk>' }
+    let(:sponsees) { ['Steve <steve@example.com>', '07700900004'] }
+
+    it 'Calls user_model#generate for the email address' do
+      expect(user_model).to have_received(:generate) \
+        .with(contact: 'steve@example.com', sponsor: 'chloe@gov.uk')
+    end
+
+    it 'Calls the user_model#generate for the phone number' do
+      expect(user_model).to have_received(:generate) \
+        .with(contact: '+447700900004', sponsor: 'chloe@gov.uk')
+    end
+
+    it 'Sends an email to the sponsee_address with the login details' do
+      expect(a_signup_email_request(email: 'steve@example.com')).to have_been_made.once
+    end
+
+    it 'Sends a sms to the sponsee_address confirming the signup' do
+      expect(a_signup_sms_request(phone_number: '+447700900004')).to have_been_made.once
+    end
+
+    context 'On production' do
+      let(:environment) { 'production' }
+      let(:plural_sponsor_confirmation_template_id) { '58e8ef4a-ca6b-40cd-81df-ec9c781fed56' }
+
+      it 'Sends a multiple user confirmation email to the sponsor' do
+        expect(a_plural_sponsor_confirmation_request).to have_been_made.once
+      end
+    end
+
+    context 'On staging' do
```

Comment:
Unclear of the purpose of this test. Is it double entry bookkeeping against the config values?

Reply:
Yep, it's asserting picking up the environment correctly.